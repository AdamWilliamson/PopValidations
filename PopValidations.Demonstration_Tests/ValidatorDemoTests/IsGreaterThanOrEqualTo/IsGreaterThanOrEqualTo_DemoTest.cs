using ApprovalTests;
using Newtonsoft.Json.Linq;
using PopValidations.Swashbuckle_Tests.Helpers;
using PopValidations_Tests.TestHelpers;
using System.Linq;
using System.Threading.Tasks;
using Xunit;

namespace PopValidations.Demonstration_Tests.ValidatorDemoTests.IsGreaterThanOrEqualTo;

public static class IsGreaterThanOrEqualTo
{
    //Begin-Request
    public record InputObject(int? NInteger);
    //End-Request

    //Begin-Validator
    public class Validator : AbstractValidator<InputObject>
    {
        public Validator()
        {
            Describe(x => x.NInteger).IsGreaterThanOrEqualTo(5);
        }
    }
    //End-Validator

    public class TestController : ControllerBase<InputObject> { }
}

public class IsGreaterThanOrEqualTo_DemoTest
{
    [Fact]
    public async Task Validation()
    {
        // Arrange
        var validationRunner = ValidationRunnerHelper.BasicRunnerSetup(new IsGreaterThanOrEqualTo.Validator());

        // Act
        var validationResult = await validationRunner.Validate(
            new IsGreaterThanOrEqualTo.InputObject(NInteger: 1)
        );

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(validationResult));
    }

    [Fact]
    public void Description()
    {
        // Arrange
        var descriptionRunner = ValidationRunnerHelper.BasicRunnerSetup(new IsGreaterThanOrEqualTo.Validator());

        // Act
        var descriptionResult = descriptionRunner.Describe();

        // Assert
        Approvals.VerifyJson(JsonConverter.ToJson(descriptionResult));
    }

    [Fact]
    public async Task OpenApi()
    {
        // Arrange
        var config = new WebApiConfig();
        var setup = new TestSetup<
               IsGreaterThanOrEqualTo.TestController,
               IsGreaterThanOrEqualTo.Validator,
               IsGreaterThanOrEqualTo.InputObject
           >();

        // Act
        var helper = await setup.GetHelper(config);
        JObject json = JObject.Parse(helper.Content!);
        var match = json["components"]!["schemas"]!.FirstOrDefault();

        // Assert
        Approvals.VerifyJson(match?.ToString(Newtonsoft.Json.Formatting.None));
    }
}