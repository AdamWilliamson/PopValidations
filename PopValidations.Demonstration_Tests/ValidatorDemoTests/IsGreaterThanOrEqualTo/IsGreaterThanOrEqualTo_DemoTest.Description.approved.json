{
    "results": [
        {
            "property": "NInteger",
            "outcomes": [
                {
                    "validator": "IsGreaterThanOrEqualToValidation",
                    "message": "Must be greater than or equal to '5'.",
                    "values": [
                        {
                            "key": "value",
                            "value": "5"
                        }
                    ]
                }
            ],
            "validationGroups": []
        }
    ]
}